crates/bevy_sprite/src/picking_backend.rs

```diff
@@ -133,14 +151,44 @@ pub fn sprite_picking(
                     .lerp(cursor_end_sprite, lerp_factor)
                     .xy();
 
-                let is_cursor_in_sprite = rect.contains(cursor_pos_sprite);
+                let Ok(cursor_pixel_space) = sprite.compute_pixel_space_point(
+                    cursor_pos_sprite,
+                    &images,
+                    &texture_atlas_layout,
+                ) else {
+                    return None;
+                };
+
+                // Since the pixel space coordinate is `Ok`, we know the cursor is in the bounds of
+                // the sprite.
+
+                let cursor_in_valid_pixels_of_sprite = 'valid_pixel: {
+                    match settings.picking_mode {
+                        SpritePickingMode::AlphaThreshold(cutoff) => {
+                            let Some(image) = images.get(&sprite.image) else {
```

Comment:
What is this the reason for returning true for an valid pixel here?

Reply:
`Sprite::from_color` returns a defaulted handle. This handle doesn't return a valid image, so returning false here would make picking "color sprites" impossible.

Reply:
Can we add a comment here?